api/project/controllers/project.js

```diff
@@ -78,5 +78,46 @@ module.exports = {
       console.log('error trying to give project google resources: ', err);
       ctx.send(err, 500);
     }
+  },
+
+  async recommendProjectToUsers(ctx) {
+    try {
+      const { interests } = ctx.state.user;
+      const projects = await strapi.services.project.find();
+      const recommendedProjectSlugs = [];
+      const recommendedProjects = [];
+
+      /*
+        This loops through the users interest's categories and compares them
+        with the project interest's categories and see if they match
+        NOTE: It pushes duplicates into the array so I prevent it from happening
+      */
+      for(const interest of interests){
+        for(const category of interest.categories){
+          for(const project of projects){
+            for(const projectInterest of project.interests){
+              for(const projectCategories of projectInterest.categories){
+                if((projectCategories.category === category.category) && (!(recommendedProjectSlugs.includes(project.slug)))){
+                  recommendedProjectSlugs.push(project.slug);
```

Comment:
I was going to do that initially, but it gave me duplicate projects since it returned a project for each category it matches, and since some projects have more than one it'll return me that project that many times. My idea was to prevent that from happening with `!(recommendedProjectSlugs.includes(project.slug)`. the includes method doesn't take in objects, so that's why I used the slugs instead.